avocado/core/task/runtime.py

```diff
@@ -112,42 +126,50 @@ def get_dependencies_form_runnable(cls, runnable, status_server_uri=None,
                        sent to the destination job's status server and will
                        make into the job's results.
         :type job_id: str
-        :returns: RUntimeTasks of the dependencies from runnable
-        :rtype: list
+        :returns: RuntimeTask of the test from runnable
         """
-        if runnable.dependencies is None:
-            return []
-
-        # creates the runnables for the dependencies
-        dependencies_runnables = DependencyResolver.resolve(runnable)
-        dependencies_runtime_tasks = []
-        # creates the tasks and runtime tasks for the dependencies
-        for dependency_runnable in dependencies_runnables:
-            name = (f"{dependency_runnable.kind}-"
-                    f"{dependency_runnable.kwargs.get('name')}")
-            prefix = 0
-            # the human UI works with TestID objects, so we need to
-            # use it to name Task
-            task_id = TestID(prefix, name)
-            # with --dry-run we don't want to run dependencies
-            if runnable.kind == 'dry-run':
-                dependency_runnable.kind = 'noop'
-            # creates the dependency task
-            dependency_task = Task(dependency_runnable,
-                                   identifier=task_id,
-                                   status_uris=status_server_uri,
-                                   category='dependency',
-                                   job_id=job_id)
-            dependencies_runtime_tasks.append(cls(dependency_task))
-
-        return dependencies_runtime_tasks
+        name = f'{pre_runnable.kind}-{pre_runnable.kwargs.get("name")}'
+        prefix = 0
+        # the human UI works with TestID objects, so we need to
+        # use it to name Task
+        task_id = TestID(prefix, name)
+        # creates the dependency task
+        task = Task(pre_runnable,
+                    identifier=task_id,
+                    status_uris=status_server_uri,
+                    category='pre_test',
+                    job_id=job_id)
+        return cls(task)
 
-    def is_dependencies_finished(self):
-        for dependency in self.dependencies:
-            if not dependency.status or not ("FINISHED" in dependency.status
-                                             or "FAILED" in dependency.status):
-                return False
-        return True
+
+def get_pre_tasks_from_runnable(runnable, status_server_uri=None, job_id=None):
```

Comment:
This method return a list of `PreRuntimeTask` objects, maybe it should be part of PreRuntimeTask as a class method.

Reply:
Yes, I was thinking about this, and I decided to go with this solution because IMO the class method should create only the object not the list of objects, But it is only matter of preferences I think.

Reply:
Being a list or a single object should not be a limiting aspect of keeping this outside the class scope. IMO, its more than a preference: it will keep the code more organized.

If you change the class name, for instance, you need to change the line 164, by using the `cls` argument, you don't.

Also, when users are looking for everything related to PreRuntimeTask the should find in a single place, if possible. I don't see a good reason here for keeping this outside the class scope.